### PR TITLE
Set ownership of proxy contracts on deployment

### PR DESCRIPTION
**Pull request by rkachowski**
Sets the ownership of two proxy contracts (SystemConfigProxy and ProtocolVersionsProxy) to the SystemOwnerSafe contract during deployment if the correct config var is set. This removes the deployment account from directly owning L1 contracts, and instead indirectly owns them through multisig membership.

closes https://github.com/celo-org/celo-blockchain-planning/issues/518

### CODE REVIEW

**Comment by palango** [comment]
I miss some context here. Do you know why this is done the way it is? For me it seems like the safe is the "better" owner. Can we we maybe upstream this?

**Comment by rkachowski** [comment]
hey @palango , I don't know why it's done this way. I would also agree that the safe is the better owner, especially in the case of Celo which doesn't have a well defined security council or ownership policy yet. I don't see why it couldn't be upstreamed

**Comment by palango** [comment]
@pahor167 @martinvol I'd like to get your input here as well.

File: packages/contracts-bedrock/scripts/deploy/ChainAssertions.sol [hunk 1]
```diff
@@ -72,7 +72,6 @@ library ChainAssertions {
         ResourceMetering.ResourceConfig memory resourceConfig = config.resourceConfig();
 
         if (_isProxy) {
-            require(config.owner() == _cfg.finalSystemOwner());
             require(config.basefeeScalar() == _cfg.basefeeScalar());
```
Comment (pahor167) [inline packages/contracts-bedrock/scripts/deploy/ChainAssertions.sol:75]: Rather then deleting this check - can we update it so it checks the owner based on config ?

File: packages/contracts-bedrock/scripts/deploy/Deploy.s.sol [hunk 2]
```diff
@@ -260,15 +260,15 @@ contract Deploy is Deployer {
         }
     }
 
-    /// @notice Transfer ownership of a Proxy to the ProxyAdmin contract
+    /// @notice Transfer admin of a Proxy to the ProxyAdmin contract
     ///         This is expected to be used in conjusting with deployERC1967ProxyWithOwner after setup actions
     ///         have been performed on the proxy.
-    /// @param _name The name of the proxy to transfer ownership of.
+    /// @param _name The name of the proxy to transfer admin of.
```
Comment (pahor167) [inline packages/contracts-bedrock/scripts/deploy/Deploy.s.sol:266]: Any reason why not to also change owner of `InitializeDataAvailabilityChallenge` ?

**Comment by pahor167** [comment]
Closed in favor of https://github.com/celo-org/optimism/pull/255

File: packages/contracts-bedrock/scripts/deploy/Deploy.s.sol (hunk 2 above)
Comment (rkachowski) [reply packages/contracts-bedrock/scripts/deploy/Deploy.s.sol:266]: iirc, this function actually changed the admin - and not owner (as described in the comment) - and was updated for the sake of clarity.

**Pull request closed without merging**